src/libsyntax/errors/emitter.rs

```diff
@@ -23,6 +22,23 @@ use std::io;
 use std::rc::Rc;
 use term;
 
+#[derive(Debug)]
+enum Error {
+    Io(io::Error),
+    Term(term::Error),
+}
+
+impl From<io::Error> for Error {
+    fn from(err: io::Error) -> Self {
+        Error::Io(err)
+    }
+}
+
+impl From<term::Error> for Error {
+    fn from(err: term::Error) -> Self {
+        Error::Term(err)
+    }
+}
```

Comment:
Yeah, I thought about that. I initially started by just adding:

```rust
impl From<term::Error> for io::Error {
    fn from(err: term::Error) -> io::Error {
        match err {
            term::Io(err) => err,
            err => io::Error::new(io::ErrorKind::Other, err),
        }
    }
}
```

But I decided against that because it seems a shame that we're giving up the flexibility to do sensible things with the `term` errors, and making it slightly harder to add in custom emitter and libtest errors in the future. There's a side of me that just wants us to not get into the habit of stuffing things into error types like `io::Error`. It probably doesn't really matter much in this case since these uses are internal to the compiler, but I'm trying to be a good role model :)